libs/coin-modules/coin-evm/src/staking/transactionData.ts

```diff
@@ -53,3 +60,53 @@ export const buildTransactionParams = (
 
   return operation(recipient, amount, sourceValidator, delegator);
 };
+
+/**
+ * Builds transaction parameters for staking transactions
+ */
+export function buildStakingTransactionParams(
+  currency: CryptoCurrency,
+  intent: AnyIntent,
+): {
+  to: string;
+  data: Buffer;
+  value: bigint;
+} {
+  if (!isStakingIntent(intent)) {
+    throw new Error("Intent must be a staking intent");
+  }
+
+  const { amount, asset, recipient, sender, mode, parameters } = intent;
+
+  const config = STAKING_CONTRACTS[currency.id];
+  if (!config) {
+    throw new Error(`Unsupported staking currency: ${currency.id}`);
+  }
+
+  if (!mode || !isStakingOperation(mode)) {
+    throw new Error(`Invalid staking operation: ${mode}`);
+  }
+
+  const stakingParams = buildTransactionParams(
+    currency.id,
+    mode,
+    recipient,
+    amount,
+    parameters?.[0], // sourceValidator for redelegate
```

Comment:
I am not sur that we need this if we only have `sourceValidator` in parameters !
we can directly have it as optional ?